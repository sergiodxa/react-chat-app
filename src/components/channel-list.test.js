import React from "react";
import { render } from "@testing-library/react";

import ChannelList from "./channel-list";

test("ChannelList component", () => {
  const { asFragment } = render(
    <ChannelList
      channels={["general", "testing", "varios"]}
      currentChannel="testing"
      onChannelChange={jest.fn()}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
