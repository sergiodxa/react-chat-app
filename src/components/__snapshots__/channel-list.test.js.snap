// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList component 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: none;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  font-size: 1rem;
  padding: .5rem;
  outline: none;
  text-align: left;
  width: 100%;
}

<div
    class="emotion-1"
  >
    <button
      aria-label="Change to channel general"
      class="emotion-0"
    >
      # general
    </button>
  </div>
  .emotion-1 {
  background-color: #783CEE;
}

.emotion-0 {
  background: none;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  padding: .5rem;
  outline: none;
  text-align: left;
  width: 100%;
}

<div
    class="emotion-1"
  >
    <button
      aria-label="Change to channel testing"
      class="emotion-0"
    >
      # testing
    </button>
  </div>
  .emotion-1 {
  background-color: none;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  font-size: 1rem;
  padding: .5rem;
  outline: none;
  text-align: left;
  width: 100%;
}

<div
    class="emotion-1"
  >
    <button
      aria-label="Change to channel varios"
      class="emotion-0"
    >
      # varios
    </button>
  </div>
</DocumentFragment>
`;
